statsmodels/sandbox/mice/mice.py

```diff
@@ -0,0 +1,1167 @@
+        h1 = ax.hist(np.asarray(obs), **obs_hist_args)
+        h2 = ax.hist(np.asarray(self.data[col_name]), **all_hist_args)
+        ha.extend([h1[-1][0], h2[-1][0]])
+        la.extend(["Obs", "All"])
+
+        leg = fig.legend(ha, la, 'center right', numpoints=1)
+        leg.draw_frame(False)
+
+        ax.set_xlabel(col_name)
+        ax.set_ylabel("Frequency")
+
+        return fig
+
+    def perturb_bootstrap(self, vname):
+        """
+        Perturbs the model's parameters using a bootstrap.
+        """
+
+        endog_obs, exog_obs, exog_miss =\
+                   self.get_split_data(vname)
+
+        m = len(endog_obs)
+        rix = np.random.randint(0, m, m)
+        endog_boot = endog_obs[rix]
+        exog_boot = exog_obs[rix, :]
+        klass = self.model_class[vname]
+        self.models[vname] = klass(endog_boot, exog_boot,
+                                  **self.init_args)
+        self.results[vname] = self.models[vname].fit(**self.fit_args)
+        self.params[vname] = self.results[vname].params
+
+    def perturb_gaussian(self, vname):
+        """
+        Perturbs the model's parameters by sampling from the Gaussian
+        approximation to the sampling distribution of the parameter
+        estimates.  Optionally, the scale parameter is perturbed by
+        sampling from its asymptotic Chi^2 sampling distribution.
+        """
+
+        endog_obs, exog_obs, exog_miss =\
+                   self.get_split_data(vname)
+
+        klass = self.model_class[vname]
+        formula = self.conditional_formula[vname]
+        self.models[vname] = klass.from_formula(formula,
+                                            self.data,
+                                            **self.init_args[vname])
+        self.results[vname] = self.models[vname].fit(
+            **self.fit_args[vname])
+
+        if self.scale_method[vname] == "fix":
+            scale_pert = 1.
+        elif self.scale_method[vname] == "perturb_chi2":
+            u = np.random.chisquare(float(self.results.df_resid))
+            scale_pert = u / float(self.results.df_resid)
+        else:
+            raise ValueError("unknown scale perturbation method")
+
+        # Can use scipy here in future
+        #from scipy.stats import multivariate_normal
+        cov = self.results[vname].cov_params().copy()
+        cov *= scale_pert
+        norm = _multivariate_normal(mean=self.results[vname].params,
+                                    cov=cov)
+        self.params[vname] = norm.rvs()
+
+    def perturb_params(self, vname):
+
+        if self.perturbation_method[vname] == "gaussian":
+            self.perturb_gaussian(vname)
+        elif self.perturbation_method[vname] == "boot":
+            self.perturb_bootstrap(vname)
+        else:
+            raise ValueError("unknown perturbation method")
+
+    def impute(self, vname):
+        # Wrap this in case we later add additional imputation
+        # methods.
+        self.impute_pmm(vname)
```

Comment:
where would the imputation method be specified, when we add alternatives to `pmm`